tests: split system test framework into standalone repository

pbrezina opened the pull request:
This change allows other project to use the framework as well.

https://github.com/SSSD/sssd-test-framework

ikerexxe reviewed (approved):
Overall LGTM, but I have a minor comment.

> https://github.com/SSSD/sssd-test-framework

I wouldn't have added the sssd prefix to the name, as it gives the idea that it can only test sssd when in reality it can be used for any IdM project testing.

pbrezina commented:
IMHO it is still strongly related to SSSD. Selected parts can be reused by other projects, but the framework as a whole aims for SSSD. And all other repositories are currently prefix with sssd.

pbrezina commented:
* `master`
  * c3a0b5f07d2a409c6bab99eeaec0ac802778f70f - tests: split system test framework into standalone repository
* `sssd-2-8`
  * 2426cb89e178e1850a724bbfa3453b05dc0d9685 - tests: split system test framework into standalone repository
